content/blog/serverless-api/index.md

```diff
@@ -0,0 +1,373 @@
+---
+title: "Scale to Zero with Serverless APIs"
+date: 2025-01-27T00:00:00
+draft: false
+meta_desc: How to run web services for practically nothing. AWS Lambda hosting for containers
+meta_image: meta.png
+authors:
+    - adam-gordon-bell
+tags:
+    - python
+    - serverless
+social:
+    twitter: >
+            🚀 Transform your low-traffic Flask APIs into cost-efficient serverless apps! Package your entire REST API as a container, deploy to AWS Lambda, and pay $0 when idle. Simple local development, standard Flask patterns, and automatic scaling - all with minimal serverless expertise needed. 
+    linkedin: >
+            Ever wondered how to run web services for practically nothing? Adam shares an approach that's perfect for side projects and low-traffic applications.
+
+            💡 The Challenge:
+            Running services 24/7 that barely get used, but need to be reliable when called.
+            🎯 The Solution:
+            Package your entire web app as a container and run it on AWS Lambda!
+            ✨ Why This Works:
+
+            Zero cost when idle - literally $0!
+            Instant activation when needed
+            Pennies per month for low traffic
+            Compatible with all web frameworks
+
+            🛠️ The Beauty of Simplicity:
+            No serverless expertise needed! Just:
+
+            Write your code normally
+            Containerize it
+            Deploy to Lambda
+            Add API Gateway
+            You're live!
+
+            💰 Real Cost Example:
+            Flask app running with:
+
+            512MB memory
+            40,000 requests/month
+            Total cost: $0.28/month (!!)
+            Zero traffic = Zero cost
+---
+Every developer has that service that needs to exist but barely gets used. Maybe it's an internal reporting API that get hourly requests but just from a single client, or maybe a side project you want hosted somewhere, but that is rarely used.
+
+These services present a unique challenge: they need to be reliable when called, but actually keeping them always on and running seems like overkill.
+<!--more-->
+{{< youtube "cjLpOJn0B6g?rel=0" >}}
+
+What if you could maintain these low-traffic services with zero cost when they're idle, while ensuring they spring to life exactly when needed? This is where AWS Lambda's container support shines. While many developers think of Lambda primarily for individual functions, it's equally capable of hosting complete REST APIs that springs to life when needed.
```

Comment:
I think a decade ago this was a pretty novel concept that you can maintain zero cost these mostly idle services. But in 2025, this is old news. I think you are going to have to have a more unique insight to serverless or come up with a different hook. Remember you are sharing it over to the serverless crowd who already knows this

Reply:
As I read the post more, I think we need to answer who is our audience. Depending on that answer, will give us slightly different editorial direction.

Reply:
The audience is python developers. I think that a Lambda Monolith, and one that works locally the same as in lambda and that uses containers instead of the zip file lambda format is a bit novel.

So this isn't the lambdas of 10 years ago, this is using lambda do host a service just like you would ECS or EKS, which to me is a lot more palatable of a workflow.

Reply:
The point I was trying to get at was, you think of lambdas as these singular functions that respond to events, but its actually a great use case for hosting full backends with several advantages.